src/save_index.cpp

```diff
@@ -405,11 +405,9 @@ void extract_summary_from_config(config& cfg_save, config& cfg_summary)
 				color_options_ = game_config::default_colors;
 				color_id_ = color_options_[color_];
 				
-				if (!u[ "color" ]. empty ()) {
-					if(u["color"].to_int()){
-						color_ = u["color"].to_int() - 1;
-						color_id_ = color_options_[color_];
-					}
+				if (u.has_attribute("color)"{
```

Comment:
...now there's a syntax error here. Otherwise, it's much better. (Also we prefer no spaces after `if`.)